src/joining.h

```diff
@@ -17,6 +17,7 @@
 /* Header-only join C++ API (high-level) */
 
 #include <limits>
+#include <iostream>
```

Comment:
It seems as though we'd want as many of the includes as possible in the `*.cu` file, rather than in the header. It didn't look like anything else in joining.h was actually using iostream

Reply:
fixed with 11a7eda